components/files/file_upload/file_upload.py

```diff
@@ -45,25 +52,66 @@ def to_json(self):
             "upload-file-size" : self.file_size,
         }
 
-def upload_file(a_file_upload_bean, a_file_rdm_bean, a_server_config, a_site_id, a_domain, a_headers):
+# Yield parts of data from file pointer until EOF. This avoids reading large files completely into memory.
+# This function is used to chunk the data if its size exceeds the maximum part size accpeted by the 
+# Sitelink3D v2 file service (and other services that accept file such as designfile)
+def read_parts(a_file_ptr, a_part_size=10485760):
+    while True:
+        part = a_file_ptr.read(a_part_size)
+        if not part:
+            break
+        yield part 
+
+def upload_file_multipart(a_server_config, a_site_id, a_file_upload_bean, media_type, encoding_type, a_headers):
+    url = "{0}/file/v1/sites/{1}/upload".format(a_server_config.to_url(), a_site_id)
+    logging.debug("Upload file to {}".format(url))
+
+    file_path = a_file_upload_bean.file_location + os.path.sep + a_file_upload_bean.file_name
+    part_index = 0
+    sitelink_max_part_size = 10485760
+    part_generator = read_parts(a_file_ptr=open(file_path, "rb"), a_part_size=sitelink_max_part_size)
+
+    sha1 = hashlib.sha1()
+    with open(file_path, "rb") as f:
+        sha1.update(f.read())
+
+    for part in part_generator:
+        file_size = os.stat(file_path).st_size
+        part_total_count = math.ceil(file_size / sitelink_max_part_size)
+
+        logging.debug("Preparing data part index {} of {} (size {})".format(part_index, part_total_count, len(part)))
+
+        form_header = {
+            "upload-uuid": str(a_file_upload_bean.upload_uuid),
+            "upload-file-name": a_file_upload_bean.file_name,
+            "upload-file-sha1": sha1.hexdigest(),

```

Comment:
please move sha1.hexdigest() outside of the for loop.

Reply:
Good find.